agent/reconciler/host_reconciler.go

```diff
@@ -156,6 +169,29 @@ func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastru
 	return ctrl.Result{}, nil
 }
 
+func (r *HostReconciler) installerController(ctx context.Context, byoHost *infrastructurev1beta1.ByoHost) (ctrl.Result, error) {
```

Comment:
This function is for all controller logic from fetching, parsing and executing the installer script. So we should keep function name as `installerController` instead of `readInstallationSecret` as it denotes a small task.

Reply:
Then maybe `executeInstallerController`?